packages/turbo-repository/rust/src/lib.rs

```diff
@@ -206,11 +242,24 @@ impl Workspace {
             .collect();
 
         // Create a ChangeMapper with no ignore patterns
-        let global_deps_package_detector =
-            GlobalDepsPackageChangeMapper::new(&self.graph, std::iter::empty::<&str>()).unwrap();
-        let mapper = ChangeMapper::new(&self.graph, vec![], global_deps_package_detector);
-        let lockfile_change = changed_lockfile.map(|s| Some(s.into_bytes()));
-        let package_changes = match mapper.changed_packages(hash_set_of_paths, lockfile_change) {
+        let change_detector = base
+            .is_some()
+            .then(|| Either::Left(DefaultPackageChangeMapperWithLockfile::new(&self.graph)))
+            .unwrap_or_else(|| Either::Right(DefaultPackageChangeMapper::new(&self.graph)));
+        let mapper = ChangeMapper::new(&self.graph, vec![], change_detector);
+
+        let lockfile_contents = if let Some(base) = base {
+            self.get_lockfile_contents(&changed_files, workspace_root, base)
+        } else if changed_files.contains(
+            AnchoredSystemPath::new(self.graph.package_manager().lockfile_name())
+                .expect("the lockfile name will not be an absolute path"),
+        ) {
+            LockfileContents::UnknownChange
+        } else {
+            LockfileContents::Unknown
```

Comment:
Same as above, we've verified the lockfile isn't in the list of changed files
```suggestion
            LockfileContents::Unchanged
```